src/lib/proof-system/zkprogram-context.ts

```diff
@@ -0,0 +1,29 @@
+import { Context } from '../util/global-context.js';
+import type { Subclass } from '../util/types.js';
+import type { ProofBase } from './proof.js';
+
+export { ZkProgramContext, DeclaredProof };
+
+type DeclaredProof = {
+  Proof: Subclass<typeof ProofBase<any, any>>;
```

Comment:
can we name this something like `ProofClass´ and `proofInstance` instead? This way its more clear which is what

Reply:
sure!